va/validation-authority.go

```diff
@@ -212,12 +208,10 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 		scheme = "https"
 		port = va.httpsPort
 	}
-	portString := strconv.Itoa(port)
-	hostPort := net.JoinHostPort(host, portString)
 
 	url := &url.URL{
 		Scheme: scheme,
-		Host:   hostPort,
+		Host:   host,
```

Comment:
I think that's a reasonable interpretation, but I think from the validation web server that we set up for testing would expect that if it's got a virtualhost on port 5001, it should expect incoming requests to have Host: <name>:5001. I'm actually slightly surprised that the Apache validation server we set up in our Python integration tests doesn't fail in this situation. Maybe it's because it set ups only a single virtualhost and it's the default?